### PR TITLE
[Multi-sig] Cancel MultiSig Motion as Owner

### LINKED ISSUE #2159: [Multi-sig] feat: cancel motion as owner
### Story

As a *user that created a multi-sig motion*, I want to be able to cancel it.

### Description

The creator can click `Cancel action` in the meatball menu
image

If the user is the creator, we call [this contract](https://github.com/JoinColony/colonyNetwork/blob/029bd755d24458a12e6dd9167d7a915992845362/contracts/extensions/MultisigPermissions.sol#L300) and the motion should immediately get cancelled due to the owner performing a cancellation request

The block-ingestor needs to catch the [`MotionCancelled`](https://github.com/JoinColony/colonyNetwork/blob/029bd755d24458a12e6dd9167d7a915992845362/contracts/extensions/MultisigPermissions.sol#L42C9-L42C24) event and update the motion state to `CANCELLED`/isFinalized to true *TBD: we need to determine if we're adding new motion types in the database, or reusing existing ones*

The widget needs to poll the motion query and change it's state to `Rejected`
image

No changes should be applied.

### Design

[Figma link](https://www.figma.com/file/SaO0Zj6e7yGqLzjEsxqVhy/Permissions?type=design&node-id=20450-35152&mode=design&t=iqzIcnoziy1lwDp6-4)

### This issue is complete when...

- [ ] The creator clicks reject, the motion should get cancelled
- [ ] Block ingestor receives the cancel event
- [ ] The changes should be reflected in the database, no changes applied in the colony

### PR DESCRIPTION
**Pull request by iamsamgibbs**
## Description

[See block-ingestor counterpart](https://github.com/JoinColony/block-ingestor/pull/229)

As a *user that created a multi-sig motion*, I want to be able to cancel it.

This PR adds a new `cancelMultiSig` saga and `MotionCanceled` event handler to block-ingestor. A new `MultiSigMeatballMenu` component has been added to the `MintTokens` completed action screen. (The same component will need adding to the other completed action screens as multisig is implemented across the other action types).

## Testing

* Step 1 - Install the multisig extension
* Step 2 - Create a new Mint Tokens action using the multisig permission type
* Step 3 - Run the following query to check the `isRejected` status of the multisig motion

```
query MyQuery {
  listColonyMultiSigs {
    items {
      action {
        type
        multiSigData {
          nativeMultiSigId
          isRejected
        }
      }
    }
  }
}

```

* Step 4 - From the completed action meatball menu, select `Reject action`
* Step 5 - Run the query again to confirm the flag has updated

## Diffs

**New stuff** ✨

* `cancelMultiSig` saga added
* `MotionCanceled` handler added
* `MultiSigMeatballMenu` added

Screenshot 2024-05-29 at 09 49 26
Screenshot 2024-05-29 at 09 49 58

Resolves #2159

### CODE REVIEW

**Review by marcocolony — APPROVED** [review]
@iamsamgibbs Nice job, everything is working well 🙌 
Export-1716978730839

**Review by bassgeta — CHANGES_REQUESTED** [review]
Great job, everything works as it should smooth as butter!
A typo is left, plus 
image
if we can make multi-sig rejected motions have the same `Failed` pill on the top left as voting reputation
image
would be perfect :ok_hand:

File: src/redux/sagas/multiSig/cancelMultiSig.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    yield waitForTxResult(txChannel);
+
+    yield put<AllActions>({
+      type: ActionTypes.MULTISIG_SET_THRESHOLDS_SUCCESS,
+      meta,
+    });
+  } catch (error) {
+    return yield putError(
+      ActionTypes.MULTISIG_SET_THRESHOLDS_ERROR,
```
Comment (jakubcolony) [inline src/redux/sagas/multiSig/cancelMultiSig.ts:40]: Same here

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Looking good apart from a few bits, nice work 👍

**Comment by iamsamgibbs** [comment]
Thanks Jakub, those should all be resolved now!

**Review by bassgeta — APPROVED** [review]
Still works like a charm, nice one :+1:

**Review by jakubcolony — APPROVED** [review]
LGTM 👍

**Pull request merged**